### DISCUSSION TITLE
Can yasa be used on a normal .py file?

### DISCUSSION THREAD

**Discussion by NhanHoThanh** [raphaelvallat/yasa#193] [Q&A] [answered]
Hi, when i stumbled across yasa, i thought it was very cool and i used it on my .py and .ipynb file, however, I got library-level exception when using yasa in my .py and ipynb file. I have tried to check for the input datatype but the problem still persist. 

When i go to the main page, i notice that yasa is a command-line libary to work with EEG. is that why i cant use it? 
Specifically, the dataset im doing is SleepEDF.

**Comment by remrama** [comment, marked as answer]
Hi @NhanHoThanh, you should be able to load any EDF file into YASA for analysis, including from a `.py` file.

> When i go to the main page, i notice that yasa is a command-line libary to work with EEG. is that why i cant use it?

Sorry, I agree this is maybe confusing. YASA is probably more of a "Python" library than a "command-line" library. It's also more designed for PSG data than EEG data (though not a huge difference). But critically, you would load your data into Python using MNE rather than YASA. YASA is for _subsequent_ steps after loading with MNE. This is how it should have worked for you in the tutorials.

Refer to this MNE documentation page for the details about importing a variety of EEG file formats: https://mne.tools/stable/auto_tutorials/io/20_reading_eeg_data.html

> Specifically, the dataset im doing is SleepEDF.

If you are using data from the [Sleep-EDF](https://physionet.org/content/sleep-edfx/) dataset, then it should work. I've used that data before and it works with YASA.

> I got library-level exception when using yasa

Can you be more specific? What was the exception you got? And what was the line of code that caused it?

Since the tutorials worked for you, I'm thinking it's just a path-related problem. Make sure that you are replacing any paths to your EDF files in the tutorials with the new paths to your local files.
